library/src/main/scala/scala/scalajs/js/special/package.scala

```diff
@@ -92,6 +92,29 @@ package object special {
   def delete(obj: scala.Any, key: scala.Any): Unit =
     throw new java.lang.Error("stub")
 
+  /** Enumerates the keys of an object.
+   *
+   *  This method is the exact equivalent of the `for (a in o) {}` statement of
+   *  JavaScript. A call of the form
+   *  {{{
+   *  forin(obj)(f)
+   *  }}}
+   *  corresponds to the JavaScript statement
+   *  {{{
+   *  for (const x in obj) {
+   *    f(x)
+   *  }
+   *  }}}
```

Comment:
This is not true if the expression `f` is side-effecting (see corresponding comment in compiler). I'm not sure how strict we are with this in comments in general (given that usual Scala semantics apply), but maybe it might be worth dropping a comment.

Reply:
Ah, right! It's maybe pedantic, but yes, I'll show the temporaries in the code examples to make it crystal clear.